### ISSUE TITLE
[studio] Bulk publishing by path without initial publish, is breaking publisher

### ISSUE THREAD

**Issue by Euquimides** [craftercms/craftercms#5745]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] The issue is in the latest released 4.0.x
- [ ] The issue is in the latest released 3.1.x

### Describe the issue

If the user bulk publish by path `/` without initial publishing, the publisher will break. Seems an issue related with permissions. 

### Steps to reproduce

Steps:

1. Create a new project.  
2. Go to `Project Tools` -> `Publishing`
3. In `Publish changes made in Studio via the UI`, add `/` to bulk publish. 
4. See the issue

### Relevant log output

```shell
[ERROR] 2022-08-23T09:53:59,298 [studio.clockTaskExecutor-39] [job.StudioPublisherTask] | Failed to publish items in site 'test-publisher' 
org.craftercms.commons.security.exception.PermissionException: Permission evaluation failed
	at org.craftercms.commons.security.permissions.annotations.HasPermissionAnnotationHandler.checkPermissions(HasPermissionAnnotationHandler.java:146) ~[crafter-commons-security-4.0.2-SNAPSHOT.jar:4.0.2-SNAPSHOT]
	at org.craftercms.commons.security.permissions.annotations.HasPermissionAnnotationHandler.checkPermissions(HasPermissionAnnotationHandler.java:95) ~[crafter-commons-security-4.0.2-SNAPSHOT.jar:4.0.2-SNAPSHOT]
	at jdk.internal.reflect.GeneratedMethodAccessor259.invoke(Unknown Source) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
	at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethodWithGivenArgs(AbstractAspectJAdvice.java:634) ~[spring-aop-5.3.22.jar:5.3.22]
	at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethod(AbstractAspectJAdvice.java:624) ~[spring-aop-5.3.22.jar:5.3.22]
	at org.springframework.aop.aspectj.AspectJAroundAdvice.invoke(AspectJAroundAdvice.java:72) ~[spring-aop-5.3.22.jar:5.3.22]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:175) ~[spring-aop-5.3.22.jar:5.3.22]
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:763) ~[spring-aop-5.3.22.jar:5.3.22]
	at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:97) ~[spring-aop-5.3.22.jar:5.3.22]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) ~[spring-aop-5.3.22.jar:5.3.22]
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:763) ~[spring-aop-5.3.22.jar:5.3.22]
	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:708) ~[spring-aop-5.3.22.jar:5.3.22]
	at org.craftercms.studio.impl.v2.service.dependency.DependencyServiceImpl$$EnhancerBySpringCGLIB$$6ff32d81.getHardDependencies(<generated>) ~[classes/:4.0.2-SNAPSHOT]
	at org.craftercms.studio.impl.v1.service.dependency.DependencyServiceImpl.getPublishingDependencies(DependencyServiceImpl.java:227) ~[classes/:4.0.2-SNAPSHOT]
	at org.craftercms.studio.impl.v1.service.dependency.DependencyServiceImpl.getPublishingDependencies(DependencyServiceImpl.java:221) ~[classes/:4.0.2-SNAPSHOT]
	at org.craftercms.studio.impl.v1.service.deployment.PublishingManagerImpl.processMandatoryDependencies(PublishingManagerImpl.java:369) ~[classes/:4.0.2-SNAPSHOT]
	at org.craftercms.studio.impl.v1.service.deployment.PublishingManagerImpl$$FastClassBySpringCGLIB$$964b4d9f.invoke(<generated>) ~[classes/:4.0.2-SNAPSHOT]
	at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218) ~[spring-core-5.3.22.jar:5.3.22]
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:793) ~[spring-aop-5.3.22.jar:5.3.22]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163) ~[spring-aop-5.3.22.jar:5.3.22]
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:763) ~[spring-aop-5.3.22.jar:5.3.22]
	at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:97) ~[spring-aop-5.3.22.jar:5.3.22]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) ~[spring-aop-5.3.22.jar:5.3.22]
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:763) ~[spring-aop-5.3.22.jar:5.3.22]
	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:708) ~[spring-aop-5.3.22.jar:5.3.22]
	at org.craftercms.studio.impl.v1.service.deployment.PublishingManagerImpl$$EnhancerBySpringCGLIB$$a7952ea8.processMandatoryDependencies(<generated>) ~[classes/:4.0.2-SNAPSHOT]
	at org.craftercms.studio.impl.v2.job.StudioPublisherTask.processPublishingRequest(StudioPublisherTask.java:296) ~[classes/:4.0.2-SNAPSHOT]
	at org.craftercms.studio.impl.v2.job.StudioPublisherTask.doPublishing(StudioPublisherTask.java:236) ~[classes/:4.0.2-SNAPSHOT]
	at org.craftercms.studio.impl.v2.job.StudioPublisherTask.executeInternal(StudioPublisherTask.java:152) ~[classes/:4.0.2-SNAPSHOT]
	at org.craftercms.studio.impl.v2.job.StudioClockTask.execute(StudioClockTask.java:69) ~[classes/:4.0.2-SNAPSHOT]
	at org.craftercms.studio.impl.v2.job.StudioClockExecutor.lambda$executeTasks$0(StudioClockExecutor.java:121) ~[classes/:4.0.2-SNAPSHOT]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[?:?]
	at java.lang.Thread.run(Thread.java:829) ~[?:?]
Caused by: org.craftercms.commons.security.exception.SubjectNotFoundException: Current subject was not found
	at org.craftercms.commons.security.permissions.impl.PermissionEvaluatorImpl.isAllowed(PermissionEvaluatorImpl.java:57) ~[crafter-commons-security-4.0.2-SNAPSHOT.jar:4.0.2-SNAPSHOT]
	at org.craftercms.commons.security.permissions.annotations.HasPermissionAnnotationHandler.checkPermissions(HasPermissionAnnotationHandler.java:144) ~[crafter-commons-security-4.0.2-SNAPSHOT.jar:4.0.2-SNAPSHOT]
	... 35 more
```

### Screenshots and/or videos

https://user-images.githubusercontent.com/90878288/186205671-6f2bb474-3ea5-4a6e-b6ea-c357e17f42da.mp4

**Comment by jmendeza** [comment]
@Euquimides please verify that solution for https://github.com/craftercms/craftercms/issues/5732  also fixes this one

**Comment by Euquimides** [comment]
Hi! Updated and rechecked this issue. Seems the solution fixed this and #5732. Thank you! Closing this ticket.
